src/Native/Runtime/amd64/PInvoke.asm

```diff
@@ -341,4 +340,59 @@ LEAF_ENTRY RhpReversePInvokeReturn, _TEXT
 LEAF_END RhpReversePInvokeReturn, _TEXT
 
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;
+;; RhpPInvoke
+;;
+;; IN:  RCX: address of pinvoke frame
+;;
+;; TRASHES: R10, R11
+;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+LEAF_ENTRY RhpPInvoke, _TEXT
+        ;; R10 = GetThread(), TRASHES R11
+        INLINE_GETTHREAD r10, r11
+
+        ;; The callsite of the helper is as good to start the stackwalk as the actual PInvoke callsite
```

Comment:
The JIT treats all helpers as GC triggering by default, so the right GC info will be produced for this one as well. The special magic needs to happen for helpers that do not trigger GC.

I have added comment about this assumption.